secp256k1-zkp-dev/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// secp256k1-zkp bindings
+// Written in 2019 by
+//   Jonas Nick
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+/// The functions in this module are copied from secp256k1 because they can only be used when
+/// compiled with the `rand` feature. But the other libraries need them as a dev-dependency for
+/// `cargo test` also when `rand` is not enabled.
+pub extern crate rand;
+pub extern crate secp256k1;
+
+use rand::Rng;
+use secp256k1::{PublicKey, Secp256k1, SecretKey, Signing};
+
+fn random_32_bytes<R: Rng>(rng: &mut R) -> [u8; 32] {
+    let mut ret = [0u8; 32];
+    rng.fill_bytes(&mut ret);
+    ret
+}
+
+trait New {
+    fn new<R: Rng>(rng: &mut R) -> SecretKey;
+}
+
+impl New for SecretKey {
+    /// Creates a new random secret key.
+    #[inline]
+    fn new<R: Rng>(rng: &mut R) -> SecretKey {
+        let mut key = SecretKey::from_slice(&random_32_bytes(rng));
+        while key.is_err() {
+            key = SecretKey::from_slice(&random_32_bytes(rng));
+        }
+        key.unwrap()
```

Comment:
It avoids an uwrap to do
```
loop {
    if let Ok(key) = SecretKey::from_slice(&random_32_bytes(rng)) {
        return key;
    }
}
```
which is a little more rustic.

Reply:
Ok (this was all just copied from rust-secp)

Reply:
fixed